File: app/src/main/java/ashcollauthlogin/darkzek/com/IntroPageTransformer.java
package ashcollauthlogin.darkzek.com;

import android.app.Activity;
import android.support.v4.view.ViewPager;
import android.view.View;

public class IntroPageTransformer implements ViewPager.PageTransformer {

    @Override
    public void transformPage(View page, float position) {


        if (page.getTag() == null) {return;}

        int pagePosition = (int) page.getTag();

        int pageWidth = page.getWidth();
        float pageWidthTimesPosition = pageWidth * position;
        float absPosition = Math.abs(position);

        // Now it's time for the effects
        if (position <= -1.0f || position >= 1.0f) {

            // The page is not visible. This is a good place to stop
            // any potential work / animations you may have running.

        } else if (position == 0.0f) {

        } else {

            View title = page.findViewById(R.id.title);
            title.setAlpha(1.0f - absPosition);

            View description = page.findViewById(R.id.description);
            if (description != null) {
                description.setTranslationY(-pageWidthTimesPosition / 2f);
                description.setAlpha(1.0f - absPosition);
            }

            View computer = page.findViewById(R.id.computer);

            if (pagePosition == 0 && computer != null) {
                computer.setAlpha(1.0f - absPosition);
                computer.setTranslationX(-pageWidthTimesPosition * 1.5f);
            }
        }
    }

}

File: app/src/main/java/ashcollauthlogin/darkzek/com/IntroPageTransformer.kt
package ashcollauthlogin.darkzek.com

import android.app.Activity
import android.support.v4.view.ViewPager
import android.view.View

class IntroPageTransformer : ViewPager.PageTransformer {

    override fun transformPage(page: View, position: Float) {


        if (page.tag == null) {
            return
        }

        val pagePosition = page.tag as Int

        val pageWidth = page.width
        val pageWidthTimesPosition = pageWidth * position
        val absPosition = Math.abs(position)

        // Now it's time for the effects
        if (position <= -1.0f || position >= 1.0f) {

            // The page is not visible. This is a good place to stop
            // any potential work / animations you may have running.

        } else if (position == 0.0f) {

        } else {

            val title = page.findViewById<View>(R.id.title)
            title.alpha = 1.0f - absPosition

            val description = page.findViewById<View>(R.id.description)
            if (description != null) {
                description.translationY = -pageWidthTimesPosition / 2f
                description.alpha = 1.0f - absPosition
            }

            val computer = page.findViewById<View>(R.id.computer)

            if (pagePosition == 0 && computer != null) {
                computer.alpha = 1.0f - absPosition
                computer.translationX = -pageWidthTimesPosition * 1.5f
            }
        }
    }

}